File: Go/011. Container With Most Water.go
func min(a int, b int) int {
	if a < b {
		return a
	}
	return b
}

func maxArea(height []int) int {
	i, j := 0, len(height) - 1
	res := 0
	for i < j {
		tmp := min(height[i], height[j]) * (j - i)
		if res < tmp {
			res = tmp
		}
		if height[i] < height[j] {
			i++
		} else {
			j--;
		}
	}
	return res
}

File: C++/011. Container With Most Water.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int maxArea(std::vector<int>& height) {
        int i = 0, j = int(height.size()) - 1;
        int res = 0;
        while (i < j) {
            int tmp = std::min(height[i], height[j]) * (j - i);
            res = std::max(res, tmp);
            if (height[i] <= height[j])
                i++;
            else
                j--;
        }
        return res;
    }
};